File: AkkaSTMExample/src/main/scala/org/akka/essentials/stm/stockticker/example/StockReader.scala
package org.akka.essentials.stm.stockticker.example
import akka.agent.Agent

class StockReader(stock: Stock) extends Runnable {

  var countDown = 10

  override def run(): Unit = {
    while (countDown > 0) {
      Thread.sleep(50)
      val x: String = Thread.currentThread().getName()
      val stockTicker = stock.price.get
      println("Quote read by thread (" + x + "), current price " + stockTicker)
      countDown = countDown - 1
    }
  }
}

File: AkkaSTMExample/src/main/java/org/akka/essentials/stm/stockticker/example/StockReader.java
package org.akka.essentials.stm.stockticker.example;

public class StockReader implements Runnable {

	private int countDown = 10;
	private Stock stock;

	public StockReader(Stock inStock) {
		stock = inStock;
	}

	public void run() {
		while (countDown > 0) {
			try {
				Thread.sleep(50);
			} catch (InterruptedException e) {
			}
			String x = Thread.currentThread().getName();
			Float stockTicker = stock.getPrice().get();
			System.out.println("Quote read by thread (" + x
					+ "), current price " + stockTicker);
			countDown = countDown - 1;
		}
	}

}
